dynawo/sources/API/PAR/PARMacroParameterSet.h

```diff
@@ -0,0 +1,82 @@
+//
+// Copyright (c) 2015-2021, RTE (http://www.rte-france.com)
+// See AUTHORS.txt
+// All rights reserved.
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, you can obtain one at http://mozilla.org/MPL/2.0/.
+// SPDX-License-Identifier: MPL-2.0
+//
+// This file is part of Dynawo, an hybrid C++/Modelica open source time domain
+// simulation tool for power systems.
+//
+
+/**
+ * @file PARMacroParameterSet.h
+ * @brief PARMacroParameterSet : interface file
+ *
+ */
+
+#ifndef API_PAR_PARMACROPARAMETERSET_H_
+#define API_PAR_PARMACROPARAMETERSET_H_
+
+#include <boost/shared_ptr.hpp>
+#include <map>
+#include <string>
+#include <vector>
+
+#include "PARReference.h"
+#include "PARParameter.h"
+
+namespace parameters {
+/**
+ * @class MacroParameterSet
+ * @brief MacroParameterSet interface class
+ */
+class MacroParameterSet {
+ public:
+  /**
+   * @brief MacroParameterSet constructor
+   * @param[in] id: id of the macroParameterSet
+   * @returns new MacroParameterSet::Impl instance with given attributes
+   */
+  explicit MacroParameterSet(const std::string& id);
+
+  /**
+   * @brief macroParameterSet id getter
+   * @returns the id of the macroParameterSet
+   */
+  const std::string& getId() const;
+
+  /**
+   * @brief reference adder
+   * @param[in] reference to be added
+   */
+  void addReference(boost::shared_ptr<Reference> reference);
+
+  /**
+   * @brief parameter adder
+   * @param[in] parameter to be added
+   */
+  void addParameter(boost::shared_ptr<Parameter> parameter);
+
+  /**
+   * @brief getter of map of references
+   * @returns reference to map of references
+   */
+  inline std::map<std::string, boost::shared_ptr<Reference> >& getReferences() {return references_;}
```

Comment:
It would be cleaner to add a begin, end and whatever operations are done  on this map rather than exposing its type and letting anyone edit it without the class control (same for all similar gets)